### PR TITLE
Uml

### PR DESCRIPTION
**Pull request by Codrux2200**
## Description
Implementation of hpp's UML

## Lien tache Jira
https://leomaman.atlassian.net/browse/TYPE-6?atlOrigin=eyJpIjoiMDk5MmZlYjRiM2FiNGYxMzgzYTExMGRiNTFjMzRlMDQiLCJwIjoiaiJ9

## Changement
ECS/*.hpp
ECS/system/*.hpp
ECS/components/*.hpp

### CODE REVIEW

File: ecs/system/AudioSystem.hpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    class AudioSystem : public ISystem{
+        public:
+            AudioSystem();
+            ~AudioSystem();
+             /**
+             * @brief init Scene Manager
+             * 
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/AudioSystem.hpp:23]: it is an override

File: ecs/system/EventSystem.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    class EventSystem : public ISystem{
+        public:
+            EventSystem();
+            ~EventSystem();
+             /**
+             * @brief init Scene Manager
+             * 
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/EventSystem.hpp:22]: it is an override

File: ecs/system/EventSystem.hpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            /**
+             * @brief update all the system
+             * 
+             * @param SceneManager 
+             * @param deltaTime 
+             */
+            void Update(SceneManager &SceneManager, int deltaTime);
```
Comment (raphaelMrci) [inline ecs/system/EventSystem.hpp:29]: it is an override

File: ecs/system/EventSystem.hpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            /**
+             * @brief destroy all the system
+             * 
+             * @param SceneManager 
+             */
+            void Destroy(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/EventSystem.hpp:35]: it is an override

File: ecs/system/GameSystem.hpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    class GameSystem : public ISystem{
+        public:
+            GameSystem();
+            ~GameSystem();
+             /**
+             * @brief init Scene Manager
+             * 
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/GameSystem.hpp:23]: it is an override

File: ecs/system/GameSystem.hpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            /**
+             * @brief update all the system
+             * 
+             * @param SceneManager 
+             * @param deltaTime 
+             */
+            void Update(SceneManager &SceneManager, int deltaTime);
```
Comment (raphaelMrci) [inline ecs/system/GameSystem.hpp:30]: it is an override

File: ecs/system/GameSystem.hpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            /**
+             * @brief destroy all the system
+             * 
+             * @param SceneManager 
+             */
+            void Destroy(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/GameSystem.hpp:36]: it is an override

File: ecs/system/GraphicSystem.hpp [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    class GraphicSystem : public ISystem{
+        public:
+            GraphicSystem();
+            ~GraphicSystem();
+             /**
+             * @brief init Scene Manager
+             * 
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/GraphicSystem.hpp:22]: it is an override

File: ecs/system/GraphicSystem.hpp [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            /**
+             * @brief update all the system
+             * 
+             * @param SceneManager 
+             * @param deltaTime 
+             */
+            void Update(SceneManager &SceneManager, int deltaTime);
```
Comment (raphaelMrci) [inline ecs/system/GraphicSystem.hpp:29]: Same thing s as other, I'm tired lol

File: ecs/system/GraphicSystem.hpp [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            /**
+             * @brief destroy all the system
+             * 
+             * @param SceneManager 
+             */
+            void Destroy(SceneManager &SceneManager);
```
Comment (raphaelMrci) [inline ecs/system/GraphicSystem.hpp:35]: Same thing s as other, I'm tired lol

File: ecs/system/iSystem.hpp [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+#ifndef ISYSTEM_HPP_
+#define ISYSTEM_HPP_
+#include "../SceneManager.hpp"
+
+namespace ECS
+{
+    enum SystemType {
+        GAME = 0,
+        GRAPHIQUE = 1,
```
Comment (raphaelMrci) [inline ecs/system/iSystem.hpp:16]: It will be better in english ???

Comment (raphaelMrci) [reply ecs/system/iSystem.hpp:16]: (GRAPHIC)

File: ecs/scene.hpp
Comment (raphaelMrci) [inline ecs/scene.hpp:1]: Rename the file

**Review by raphaelMrci — CHANGES_REQUESTED** [review]
Rename files, be careful with namespaces, be CamelCase compliant, be careful when overriding virtual methods by specifying you are overriding them, write clean code and it should be good

File: CMakeLists.txt [hunk 12]
```diff
@@ -9,6 +9,6 @@ find_package(asio REQUIRED)
 
 add_executable(MySFMLProject main.cpp)
 
-target_include_directories(MySFMLProject PRIVATE ${SFML_INCLUDE_DIRS} ${ASIO_INCLUDE_DIRS})
+target_include_directories(MySFMLProject PRIVATE ${SFML_INCLUDE_DIRS} ${ASIO_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR}/ecs)
```
[the remark this answers is not in the document]
Comment (mangasteak) [reply CMakeLists.txt:12]: il me semble que tu n'as pas a inclure les hpp nn ?

File: ecs/components/healthComponent.hpp [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+** Rtype
+** File description:
+** healthComponent
+*/
+
+#ifndef HEALTHCOMPONENT_HPP_
+#define HEALTHCOMPONENT_HPP_
+#include "Acomponent.hpp"
+
+class healthComponent : public ECS::Acomponent {
```
[the remark this answers is not in the document]
Comment (mangasteak) [reply ecs/components/healthComponent.hpp:12]: is it realy necessary for class name ?

File: ecs/system/EventSystem.hpp [hunk 14]
```diff
@@ -0,0 +1,41 @@
+/*
+** EPITECH PROJECT, 2023
+** RtypeExperimentation
+** File description:
+** EventSystem
+*/
+
+#ifndef PARTICLESYSTEM_HPP_
+#define PARTICLESYSTEM_HPP_
+#include "iSystem.hpp"
```
Comment (mangasteak) [inline ecs/system/EventSystem.hpp:10]: Upercase at the beggining for FileName

**Review by mangasteak — CHANGES_REQUESTED** [review]
Raphael did review the most things, nothing to serious , just be carefull with the names of class and files

File: ecs/components/healthComponent.hpp (hunk 13 above)
[the remark this answers is not in the document]
Comment (raphaelMrci) [reply ecs/components/healthComponent.hpp:12]: > is it realy necessary for class name ?

Especially for class names

[the remark this answers is not in the document]
Comment (raphaelMrci) [reply ecs/components/healthComponent.hpp:12]: I prefer the class to be outside the namespace, than that ! It is the basics of CamelCase

File: ecs/components/ShootComponent.hpp [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+     * 
+     */
+    class ShootComponent : public ECS::AComponent {
+        public:
+            /**
+             * @brief Construct a new shoot Component object
+             * 
+             * @param uid 
+             */
+            ShootComponent(int uid, int shootSpeed, int shootDamage);
```
Comment (mangasteak) [inline ecs/components/ShootComponent.hpp:24]: should first letter of constructor / destructor be in Maj ?

File: ecs/system/ParticleSystem.hpp [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager) override;
+            /**
+             * @brief update all the system
+             * 
+             * @param SceneManager 
+             * @param deltaTime 
+             */
+            void Update(SceneManager &SceneManager, int deltaTime) override;
```
Comment (mangasteak) [inline ecs/system/ParticleSystem.hpp:41]: why does the Caps of the first letter of functions are irregular ?

**Review by mangasteak — CHANGES_REQUESTED** [review]

File: ecs/components/ShootComponent.hpp (hunk 15 above)
Comment (christophechr) [reply ecs/components/ShootComponent.hpp:24]: Constructor and destructor mostly in caps following the class case style.

File: ecs/Core.hpp [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+** Rtype
+** File description:
+** Core
+*/
+
+#ifndef CORE_HPP_
+#define CORE_HPP_
+#include <iostream>
+#include <map>
+#include "./system/ISystem.hpp"
```
Comment (mangasteak) [inline ecs/Core.hpp:12]: relative path have to be resolved with CMAKE

File: ecs/system/AudioSystem.hpp [hunk 18]
```diff
@@ -0,0 +1,91 @@
+/*
+** EPITECH PROJECT, 2023
+** RtypeExperimentation
```
Comment (mangasteak) [inline ecs/system/AudioSystem.hpp:3]: remove the experimentation from the header :)

**Review by mangasteak — CHANGES_REQUESTED** [review]

File: ecs/IComponent.hpp [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+             * @return ComponentType
+             */
+            virtual void setUid(int uid) = 0;
+        protected:
+        private:
+            /**
+             * @brief define the type of the component
+             * 
+             */
+            int _uid;
```
Comment (raphaelMrci) [inline ecs/IComponent.hpp]: No declaration of things in interfaces

File: ecs/Scene.hpp [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+             * 
+             */
+            SceneType sceneType;
+        private:
+            /**
+             * @brief load entities from the specified string
+             * 
+             * @param entityName 
+             */
+            void loadEntity(std::string entityName);
```
Comment (raphaelMrci) [inline ecs/Scene.hpp:48]: underscores

Comment (raphaelMrci) [reply ecs/Scene.hpp:48]: apply on other variables

File: ecs/system/AudioSystem.hpp [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+             * @brief Destroy the Audio System object
+             * 
+             */
+            ~AudioSystem();
+             /**
+             * @brief init Scene Manager
+             * 
+             * @param SceneManager 
+             */
+            void init(SceneManager &SceneManager) override;
```
Comment (raphaelMrci) [inline ecs/system/AudioSystem.hpp:35]: Lowercase for method params PLSSSS

Comment (raphaelMrci) [reply ecs/system/AudioSystem.hpp:35]: Valable for rest of the PR

**Review by raphaelMrci — CHANGES_REQUESTED** [review]

**Review by raphaelMrci — APPROVED** [review]
Why do you use `init` and `destroy` methods instead of using Constructors and Destroyers ? @Codrux2200

**Review by raphaelMrci — APPROVED** [review]

**Review by mangasteak — APPROVED** [review]

**Pull request merged**